kitsune/search/v2/es7_utils.py

```diff
@@ -30,52 +38,51 @@ def mapping_func(position_filter_tuple):
     return list(map(mapping_func, enumerate(filter_list)))
 
 
-def _get_locale_specific_analyzer(locale):
-    """Get an analyzer for locales specified in config otherwise return `None`"""
-
-    locale_analyzer = config.ES_LOCALE_ANALYZERS.get(locale)
-
-    if not locale_analyzer:
-        return None
-
-    if locale_analyzer.get("plugin") and not settings.ES_USE_PLUGINS:
-        return None
-
-    # use default values from ES_DEFAULT_ANALYZER if not overridden
-    locale_analyzer = config.ES_DEFAULT_ANALYZER | locale_analyzer
-
-    # turn dictionaries into `char_filter` and `token_filter` instances
-    locale_analyzer["filter"] = _insert_custom_filters(locale, locale_analyzer["filter"])
-    locale_analyzer["char_filter"] = _insert_custom_filters(
-        locale, locale_analyzer["char_filter"], char=True
-    )
-
-    return analyzer(
-        locale,
-        tokenizer=locale_analyzer["tokenizer"],
-        filter=locale_analyzer["filter"],
-        char_filter=locale_analyzer["char_filter"],
+def _create_synonym_graph_filter(synonym_file_name):
+    filter_name = f"{synonym_file_name}_synonym_graph"
+    return token_filter(
+        filter_name,
+        type="synonym_graph",
+        synonyms_path=f"synonyms/{synonym_file_name}.txt",
+        # we must use "true" instead of True to work around an elastic-dsl bug
+        lenient="true",
+        updateable="true",
     )
 
 
-def es_analyzer_for_locale(locale):
+def es_analyzer_for_locale(locale, search_analyzer=False):
     """Pick an appropriate analyzer for a given locale.
     If no analyzer is defined for `locale` or the locale analyzer uses a plugin
-    but using plugin is turned off from settings, return ES analyzer named "standard".
+    but using plugin is turned off from settings, return an analyzer named "default_sumo".
     """
 
-    local_specific_analyzer = _get_locale_specific_analyzer(locale=locale)
+    name = locale
```

Comment:
In the code below both `name` and `locale` is used. Maybe it's clearer to default `name` to empty string and use it only when it evaluates to True. In all other places we can use the `locale` (where applicable)